### PR TITLE
Restructure item browse and search results

### PR DESCRIPTION
**Pull request by matt-bernhardt**
## Developer

This restructures how we display item browse and search result pages:

1. We are adding a fourth element to both displays, the "resource class" for item records.
2. The DOM order of the four elements is changing to be more comprehensible for screen readers and crawlers: Title, Class, Description, Image (instead of Image, Title, Description).
3. To make this shared logic more sustainable, we abstract out a new partial which gets called by both displays. Part of this abstraction involves introducing the theme's first helper class, which requires a change to the Makefile in order to deploy correctly.
5. The UI of the search results page is tweaked

### Tickets affected

- https://mitlibraries.atlassian.net/browse/post-88 (adding resource class)
- https://mitlibraries.atlassian.net/browse/post-184 (update DOM order)
- https://mitlibraries.atlassian.net/browse/post-161 (update search UI)

### Version (see config/theme.ini)

- [ ] The theme's version number has been incremented, setting up a production
      deploy.

### Documentation

- [ ] Project documentation has been updated.
- [x] No documentation changes are needed.

### Accessibility

- [x] ANDI or WAVE has been run in accordance to [our guide](https://mitlibraries.github.io/guides/basics/a11y.html).
- [ ] New flagged issues not already resolved in this PR have been ticketed 
      (link in the Pull Request details above).
- [ ] This PR contains no changes to the view layer.

### Stakeholder approval

- [x] Stakeholder approval has been confirmed (see ticket).
- [ ] Stakeholder approval will happen retroactively.
- [ ] Stakeholder approval is not needed.

### Dependencies

- [ ] New dependencies have been added
- [ ] Dependencies have been updated
- [x] No dependencies have changed

### Additional context needed to review

LInks and screenshots can be found in https://mitlibraries.atlassian.net/browse/post-88

##### Code

- [x] I have confirmed that the code works as intended.
- [x] Any CodeClimate issues have been fixed or confirmed as
      added technical debt.
- [x] New dependencies are appropriate or there were no changes.

##### Documentation

- [x] The commit message is clear and follows our guidelines
      (not just this pull request message).
- [x] The documentation has been updated or is unnecessary.

### CODE REVIEW

**Comment by matt-bernhardt** [comment]
@vab , I think Infra got tagged because this PR changes the Makefile (because we're adding a new code folder - helpers - that needs to get deployed). Thanks for the quick review! Once EngX weighs in on the view changes, I'll get this merged.

**Review by cabutlermit — APPROVED** [review]
@matt-bernhardt I'm pretty sure that Infra got tagged on this PR because of the change to the `Makefile`. Apparently, there is a new `helper` folder in the theme that needs to be managed. The additional command to copy that folder and its files looks correct to me.

File: helper/ResourceClasses.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+namespace OmekaTheme\Helper;
+
+use Laminas\View\Helper\AbstractHelper;
+
+class ResourceClasses extends AbstractHelper
+{
+
+    /**
+     * Returns a representation of resource classes in HTML.
+     *
```
Comment (JPrevost) [inline helper/ResourceClasses.php:11]: Non-blocking: As there are more protection lines than happy path lines of code in this function, would you mind adding to the function docs here as to what it protects against? `* returns and empty string if supplied with no resource, and empty resource, or a resources without ID` [I'm not sure if that's true but that's what I think is happening]

**Review by JPrevost — APPROVED** [review]
Thanks for the links to allow viewing the changes more easily!

I only had non-blocking feedback so I'm approving this as-is.

**Comment by matt-bernhardt** [comment]
It looks like there is a comment syntax available in `theme.ini`, so I've added three blocks with references to where those sections are described in the Omeka docs. I've also added a line in the helper method along the lines of what you're describing. Thanks!

**Pull request merged**